src/commands/exec.ts

```diff
@@ -189,15 +189,18 @@ export default class Exec extends BaseCommand {
             process.exit(error_code);
           }
         } else {
-          return done(new ArchitectError(`Unknown stream type: ${stream_num}`));
+          // If we get an unrecognized stream number, continue outputting to stdout.
+          // This can happen when writing binary data, and kubectl seems to handle this by just
+          // writing anyways so we do the same.
+          process.stdout.write(buffer);
```

Comment:
This was a bit of a bonus discovery - I was testing with massive binary files to reproduce this (e.g. `head -c10000000 /dev/urandom > junk.txt`) and `architect exec ... -- cat junk.txt` would fail for some of these files due to `Unknown stream type`. What was happening is we were getting a unicode replacement character `0xef 0xbf 0xbd` followed by `0x01` (expected for stdout) and the rest of the line. Not exactly sure what the deal was with that, but kubectl has no problem cat-ing this entire file, so writing the buffer to stdout even if it's invalid seems to mimic kubectl behavior